tests/core/test_common.py

```diff
@@ -179,6 +180,84 @@ def test_collector(self):
             tag = "check:%s" % check.name
             assert tag in all_tags, all_tags
 
+    def test_announce_topology_data_presense(self):
+        self.setUpAgentCheck()
+
+        instance_key = {
+            "type": "type",
+            "url": "http://localhost:5050"
+        }
+        hashable_key = tuple(sorted(instance_key.items()))
+
+        self.ac.component(instance_key, "test-component1", {"name": "container"}, {"tags": ['tag1', 'tag2']})
+        self.assertEquals(len(self.ac.topology_instances), 1)
+        self.assertEquals(len(self.ac.topology_instances[hashable_key]._components), 1)
+        self.assertEquals(len(self.ac.topology_instances[hashable_key]._relations), 0)
+        expected_component_1 = {"externalId": "test-component1", "type": {"name": "container"}, "data": {"tags":['tag1', 'tag2']}}
+        self.assertEquals(self.ac.topology_instances[hashable_key]._components[0], expected_component_1)
```

Comment:
Some comments about what is being tested would be very nice (for the entire test)